tests/core/runners/test_ase.py

```diff
@@ -47,6 +52,196 @@ def teardown_function():
+
+
+def test_geodesic_interpolate_wrapper_large_system(setup_test_environment):
+    rng = np.random.default_rng()  # Create a random number generator instance
+    large_atoms = Atoms("H" * 40, positions=rng.random((40, 3)))
+
+    # Test with large system to trigger sweeping updates
+    smoother_path = _geodesic_interpolate_wrapper(large_atoms, large_atoms)
+    assert len(smoother_path) == 20
+
+
+@pytest.mark.parametrize(
+    (
+        "method",
+        "optimizer_class",
+        "precon",
+        "n_intermediate",
+        "k",
+        "max_steps",
+        "fmax",
+        "expected_logfile",
+        "r_positions",
+        "p_energy",
+        "first_image_forces",
+        "second_images_positions",
+        "index_ts",
+        "pot_energy_ts",
+        "forces_ts",
+        "last_images_positions",
+    ),
+    [
+        # ("aseneb", NEBOptimizer, None, 10, 0.1, 3, 1e-3, None),
+        # ("aseneb", SciPyFminBFGS, None, 1000, 0.1, 3, 1e-3, "some_logdir",
+        #   0.78503956131, -24.9895786292, -0.0017252843, 0.78017739462, 9, -19.946616164,
+        #   -0.19927549, 0.51475535802),
+        (
+            "aseneb",
+            NEBOptimizer,
+            None,
+            10,
+            0.1,
+            3,
+            1e-3,
+            "some_logdir",
+            -0.854,
+            1.082,
+            -0.005,
+            -0.8161139,
+            9,
+            -19.946616164,
+            -0.19927549,
+            0.51475535802,
+        )
+    ],
+)
+def test_run_neb(
+    setup_test_environment,
+    tmp_path,
+    method,
+    optimizer_class,
+    precon,
+    n_intermediate,
+    k,
+    max_steps,
+    fmax,
+    expected_logfile,
+    r_positions,
+    p_energy,
+    first_image_forces,
+    second_images_positions,
+    index_ts,
+    pot_energy_ts,
+    forces_ts,
+    last_images_positions,
+):
+    reactant, product = setup_test_environment
+
+    optimized_r = strip_decorator(relax_job)(reactant)["atoms"]
+    optimized_p = strip_decorator(relax_job)(product)["atoms"]
+
+    optimized_r.calc = EMT()
+    optimized_p.calc = EMT()
+
+    images = _geodesic_interpolate_wrapper(
+        optimized_r.copy(), optimized_p.copy(), nimages=n_intermediate
+    )
+    for image in images:
+        image.calc = EMT()
+    assert optimized_p.positions[0][1] == pytest.approx(-0.192, abs=1e-2)
+    assert optimized_r.positions[0][1] == pytest.approx(r_positions, abs=1e-2)
+    assert optimized_p.get_potential_energy() == pytest.approx(
+        p_energy, abs=1e-2
+    ), "pdt pot. energy"
+    assert optimized_p.get_forces()[0, 1] == pytest.approx(
+        first_image_forces, abs=1e-3
+    ), "pdt forces"
+
+    neb_kwargs = {"method": "aseneb", "precon": None}
+    dyn = run_neb(images, optimizer=NEBOptimizer, neb_kwargs=neb_kwargs)
+    neb_summary = summarize_neb_run(dyn)
+
+    assert neb_summary["trajectory_results"][1]["energy"] == pytest.approx(
+        1.098, abs=0.01
+    )
```

Comment:
A more thorough test of the output schema is required.

Reply:
Same problem as the geodesic here because I am using geodesic as the precursor to this step.

Reply:
There are too many things being tested here at once is the major problem. For instance, `runners/test_ase.py` should be testing the runner: `run_neb`. It should be asserting that `run_neb()` functions appropriately and nothing else. There should be no precursor to that step and nothing afterwards (i.e. no summarizing).

You'd then make a separate test for summarizing (e.g. `tests.cores.schemas`) and so on and so forth.